src/components/forms/formComponents.js

```diff
@@ -0,0 +1,387 @@
+      }
+    ]
+    if type is 'string', then a textbox is rendered
+    when checked.
+  */
+
+  const {
+    id, disabled, error,
+    formValue, options, ariaDescribedby,
+    setError
+  } = config;
+
+  return div({},
+    [
+      div(
+        {
+          className: 'radio-group',
+          id: id,
+        },
+        options.map(((option, idx) => {
+          const optionId = (!isNil(option.id) ? option.id : option.name);
+
+          return div({
+            key: idx,
+            className: 'radio-button-container',
+          }, [
+            h(RadioButton, {
+              id: `${id}_${optionId}`,
+              name: `${id}_${optionId}`,
+              key: idx,
+              defaultChecked: !isNil(formValue) && formValue.selected === option.name,
+              onClick: () => {
+                if (option.type === 'string') {
+                  onFormInputChange(config, {
+                    selected: option.name,
+                    value: '',
+                  });
+                  return;
+                }
+
+                onFormInputChange(config, { selected: option.name });
+              },
+              style: {
+                fontFamily: 'Montserrat',
+                fontSize: '14px',
+              },
+              description: option.text,
+              disabled,
+            }),
+            input({
+              isRendered: option.type === 'string' && formValue.selected === option.name,
+              id: `${id}_${optionId}_text_input`,
+              type: 'text',
+              className: `form-control ${error ? 'errored' : ''}`,
+              placeholder: option.placeholder,
+              value: formValue.value,
+              style: {
+                ...styles.inputStyle,
+                ...{
+                  marginTop: '0.5rem',
+                }
+              },
+              disabled: disabled,
+              onChange: (event) => onFormInputChange(config, {
+                selected: option.name,
+                value: event.target.value,
+              }),
+              onFocus: () => setError(),
+              onBlur: (event) => validateFormInput(config, event.target.value),
+              'aria-describedby': ariaDescribedby,
+            }),
+          ]);
+        }))
+      )
+    ]
+  );
+};
+
+export const formInputCheckbox = (config) => {
+  const {
+    id, disabled, error, toggleText,
+    formValue, ariaDescribedby
+  } = config;
+
+  return div({ className: 'checkbox' }, [
+    input({
+      type: 'checkbox',
+      id: `cb_${id}_${toggleText}`,
+      checked: formValue,
+      className: 'checkbox-inline',
+      'aria-describedby': ariaDescribedby,
+      onChange: (event) => onFormInputChange(config, event.target.checked),
+      disabled
+    }),
+    label({
+      className: `regular-checkbox ${error ? 'errored' : ''}`,
+      htmlFor: `cb_${id}_${toggleText}`,
+    }, [toggleText])
+  ]);
+};
+
+export const formInputSlider = (config) => {
```

Comment:
Not necessarily a fix for this PR, but the public visibility input slider has a default of true, but onChange isn't being called until the user interacts with the slider (so it is not initially being set in the formData). Since this is a required boolean value, I would expect the backend to store it as false if no value is sent over in the formData json

Reply:
Good to know! The mockups show this as a default true value (even tho it doesnt match the back end), maybe when we define the form data, that's where we should set it instead!